### PR TITLE
Release AttributeCachingFileSystem 1.0.0

### PR DESCRIPTION
**Pull request by mbayerPK**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,18 +4,29 @@ A java based file attribute caching filesystem library
...
+        mavenCentral()
+    }
+}
+```
+
+Dependency to add to your top level `build.gradle.kts` file:
 ```kotlin
 dependencies {
+    // If you want to use the attribute caching filesystem for both running code and test code use it this way.
     implementation("com.pkware.filesystem:file-attribute-caching:VERSION_HERE")
```
[the remark this answers is not in the document]
Comment (mbayerPK) [reply README.md:24]: sorry im not understanding what you are meaning here? I included both lines and the comments say to use one or the other depending on if you need the caching for testing and runnable code or just testing

File: README.md [hunk 2]
```diff
@@ -4,18 +4,29 @@ A java based file attribute caching filesystem library
...
+Ensure your `settings.gradle.kts` file can pull dependencies from maven central.
+```kotlin
+dependencyResolutionManagement {
+    repositories {
+        mavenCentral()
+    }
+}
+```
+
+Dependency to add to your top level `build.gradle.kts` file:
```
Comment (PKJosh) [inline README.md:20]: This is out of scope for documentation. I think supplying something like 

```
implementation("com.pkware.filesystem:file-attribute-caching:VERSION_HERE")
testImplementation("com.pkware.filesystem:file-attribute-caching:VERSION_HERE")
```

is fine for a quick copy/paste, but it's not our role to document Gradle, let alone such a basic function of Gradle.

File: README.md [hunk 3]
```diff
@@ -56,4 +67,24 @@ val cachingPath = cachingFilesystem.convertToCachingPath(path)
 // Continue your code using cachingPath...
 ```
 
+## Releasing:
+1. Make and checkout a release branch on github.
+2. Change the version in gradle.properties to a non-SNAPSHOT version.
+3. Update the CHANGELOG.md for the impending release.
+4. Run `git commit -am "Release AttributeCachingFileSystem X.Y.Z."` (where X.Y.Z is the new version) in the terminal or
```
Comment (PKJosh) [inline README.md:74]:
```suggestion
4. Run `git commit -am "Release X.Y.Z."` (where X.Y.Z is the new version) in the terminal or
```

File: README.md [hunk 4]
```diff
@@ -56,4 +67,24 @@ val cachingPath = cachingFilesystem.convertToCachingPath(path)
...
+   command line.
+5. Make a PR with your changes.
+6. Merge the release PR after approval, tag the commit on the main branch with
+   `git tag -a X.Y.Z -m "AttributeCachingFileSystem X.Y.Z"`(X.Y.Z is the new version).
```
Comment (PKJosh) [inline README.md:78]:
```suggestion
   `git tag -a X.Y.Z -m "X.Y.Z"`(X.Y.Z is the new version).
```

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mbayerPK) [reply README.md:24]: ahh we talked about this and Im going to leave it as is since i have a comment about changing the version and VERSION_HERE will prompt the user to change the version to the most updated one rather than having the dev update it each time we do a new release

File: README.md (hunk 3 above)
Comment (mbayerPK) [reply README.md:74]: I was just following the conventions I did for my detekt extensions and for our own internal projects though

File: README.md (hunk 4 above)
Comment (mbayerPK) [reply README.md:78]: I was just following the conventions I did for my detekt extensions and for our own internal projects though

File: README.md (hunk 3 above)
Comment (PKJosh) [reply README.md:74]: We do this for core-libraries because we release multiple projects from it. It doesn't make a lot of sense for something like this. I didn't think about it at the time for [DetektExtensions](https://github.com/pkware/detektExtensions/pull/2#discussion_r897218763) because I was caught up on the other use of copy/paste I disagreed wiith.

File: README.md (hunk 2 above)
Comment (mbayerPK) [reply README.md:20]: these sections im going to keep as this is open source, they are small yet helpful additions and I dont see these instructions needing to be updated anytime soon

File: README.md (hunk 3 above)
Comment (mbayerPK) [reply README.md:74]: okay i will do this

File: README.md (hunk 4 above)
Comment (mbayerPK) [reply README.md:78]: okay i will do this

File: README.md [hunk 5]
```diff
@@ -4,18 +4,29 @@ A java based file attribute caching filesystem library
...
+    }
+}
+```
+
+Dependency to add to your top level `build.gradle.kts` file:
 ```kotlin
 dependencies {
+    // If you want to use the attribute caching filesystem for both running code and test code use it this way.
     implementation("com.pkware.filesystem:file-attribute-caching:VERSION_HERE")
+    // If you to only use the attribute caching filesystem for testing include it this way.
```
Comment (MatthewFoster624) [inline README.md:25]:
```suggestion
    // To use the attribute caching filesystem for testing only:
```

File: README.md [hunk 6]
```diff
@@ -4,18 +4,29 @@ A java based file attribute caching filesystem library
...
+    repositories {
+        mavenCentral()
+    }
+}
+```
+
+Dependency to add to your top level `build.gradle.kts` file:
 ```kotlin
 dependencies {
+    // If you want to use the attribute caching filesystem for both running code and test code use it this way.
```
Comment (MatthewFoster624) [inline README.md:23]:
```suggestion
    // To use the attribute caching filesystem for both production and testing:
```

File: README.md (hunk 2 above)
Comment (MatthewFoster624) [reply README.md:20]: In light of this being a public / open source repo, I think concise documentation is better.
In this case, concise means specific to this code. For example, lines 11-19 are not.

Comment (mbayerPK) [reply README.md:20]: ehh im still against that, as standalone instructions for a standalone open source thing I would assume someone coming for this would want basic instructions of how to install it/use it

i have seen too many public repos that dont have any instructions for installation and then you have to go through maven/gradle docs- which are by no means simple to do for a new user

i dont think theres anything confusing about whats listed even though its not project specific this project will be published on maven rather than google or another public publisher

Comment (PKJosh) [reply README.md:20]: It might even be worth something like a badge notation, such as those used on https://github.com/apache/logging-log4j2. Check out (https://shields.io/category/dependencies). This is more usable and provides guidance on _multiple_ languages rather than going into detail of only one.

Comment (PKJosh) [reply README.md:20]: > ehh im still against that, as standalone instructions for a standalone open source thing I would assume someone coming for this would want basic instructions of how to install it/use it
> 
> i have seen too many public repos that dont have any instructions for installation and then you have to go through maven/gradle docs- which are by no means simple to do for a new user

But our documentation isn't geared towards teaching you how to import dependencies. Nobody seeking use of our library is at their first idiomatic rodeo.

> i dont think theres anything confusing about whats listed even though its not project specific this project will be published on maven rather than google or another public publisher

It's about brevity. We don't teach them where to download a JVM, or how to include Kotlin in their project. Why? Because it's out of scope.

Things that are more relevant for documentation would be educating someone on using this from _Java_, which is not covered.

Comment (mbayerPK) [reply README.md:20]: Since when do I have to apply the assumption about what the target audience is for an open source library? we dont have any standards for this as a company currently.

Ideally as a developer looking at a third party library I would want the following from a library readme at the very least:

- installation instructions for the library in the language it's written in along with the source its being published from
- usage instructions in the language it was written in
- release instructions for the repo it is stored on

anything else is a bonus- none of the other projects we have go into using different languages unless its an official sdk or the like where that is expected

Comment (PKJosh) [reply README.md]: > Since when do I have to apply the assumption about what the target audience is for an open source library? we dont have any standards for this as a company currently.

Almost every time you write javadocs, I remind you that the target audience is an advanced (or at the very least intermediate) developer. 

If I'm evaluating a 3rd-party tool, I highly value good documentation and good tests. As @MatthewFoster624 alluded to, part of good documentation is providing only the necessary.

You're saying we don't have a standard, and I'm trying to establish one based on how the better open source projects tend to write their documentation, which educates users on _their_ offering. This isn't an exhaustive how-to implement with our code from start to finish, it's an opportunity to give a quick overview things developers typically look for.

I was on the fence about suggesting moving the `Release` instructions to `RELEASING.MD` for that very reason, but did not.

> anything else is a bonus- none of the other projects we have go into using different languages unless its an official sdk or the like where that is expected

Right, so are you saying that educating someone on how to use Java from Kotlin is out of scope? So is importing `mavenCentral` in Gradle. But educating on how to use this library from Java is FAR more helpful documentation than a trivial Gradle command.

I know you think you're providing a bonus, but you're mostly insulting the reader IMO.

Comment (PKJosh) [reply README.md]: I should take the time to say that I think your initial documentation does a fantastic job of what I'm advocating on behalf of!

File: README.md (hunk 5 above)
Comment (MatthewFoster624) [reply README.md]: @mbayerPK  This didn't make it in

File: README.md (hunk 6 above)
Comment (MatthewFoster624) [reply README.md]: @mbayerPK  This didn't make it in

**Review by MatthewFoster624 — APPROVED** [review]

**Review by kirale18 — APPROVED** [review]
